cpp/src/arrow/adapters/orc/adapter.cc

```diff
@@ -80,6 +74,12 @@ namespace liborc = orc;
   }                                            \
   catch (const liborc::NotImplementedYet& e) { \
     return Status::NotImplemented(e.what());   \
+  }                                            \
+  catch (const std::exception& e) {            \
+    return Status::Invalid(e.what());          \
+  }                                            \
+  catch (...) {                                \
+    return Status::UnknownError("ORC error");  \
```

Comment:
Do we need this? (Is `catch (const std::exception& e)` enough?)

Reply:
Usually it is enough. But just in case?